presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PushAggregationThroughOuterJoin.java

```diff
@@ -249,35 +254,60 @@ private Optional<PlanNode> coalesceWithNullAggregation(AggregationNode aggregati
         AggregationNode aggregationOverNull = aggregationOverNullInfo.getAggregation();
         Map<VariableReferenceExpression, VariableReferenceExpression> sourceAggregationToOverNullMapping = aggregationOverNullInfo.getVariableMapping();
 
-        // Do a cross join with the aggregation over null
-        JoinNode crossJoin = new JoinNode(
-                outerJoin.getSourceLocation(),
-                idAllocator.getNextId(),
-                JoinNode.Type.INNER,
-                outerJoin,
-                aggregationOverNull,
-                ImmutableList.of(),
-                ImmutableList.<VariableReferenceExpression>builder()
-                        .addAll(outerJoin.getOutputVariables())
-                        .addAll(aggregationOverNull.getOutputVariables())
-                        .build(),
-                Optional.empty(),
-                Optional.empty(),
-                Optional.empty(),
-                Optional.empty(),
-                ImmutableMap.of());
+        FunctionResolution functionResolution = new FunctionResolution(functionAndTypeManager.getFunctionAndTypeResolver());
+        Map<VariableReferenceExpression, RowExpression> literalMap = new HashMap<>();
+
+        if (useDefaultsForCorrelatedAggregations) {
+            for (Map.Entry<VariableReferenceExpression, AggregationNode.Aggregation> aggregation : aggregationNode.getAggregations().entrySet()) {
+                FunctionHandle functionHandle = aggregation.getValue().getFunctionHandle();
+
+                Optional<RowExpression> defaultLiteral = Optional.empty();
+                if (functionResolution.isCountFunction(functionHandle)) { // Can also include count_if
```

Comment:
can we look specifically for count(value).  i don't want to get wrong results if we start to support count(*)